src/controller/property.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  Property,
+  saveProperty,
+  findProperty,
+  updateProperty,
+  removeProperty
+} from "./../models/property";
+import { Response, Request, NextFunction } from "express";
+import { validationResult } from "express-validator/check";
+
+export function postProperty(req: Request, res: Response, next: NextFunction) {
+  const errors = validationResult(req);
+
+  if (!errors.isEmpty()) {
+    return res.status(422).json({ errors: errors.array() });
+  }
+
+  const property = req.body as Property;
+
+  return saveProperty(property)
+    .then(property => {
+      return res.json({
+        property
+      });
+    })
+    .catch(next);
+}
+
+export function putProperty(req: Request, res: Response, next: NextFunction) {
+  const errors = validationResult(req);
+
+  if (!errors.isEmpty()) {
+    return res.status(422).json({ errors: errors.array() });
+  }
+
+  const property = req.body as Property;
+
+  return updateProperty(property)
+    .then(property => {
+      return res.json({
+        property
+      });
+    })
+    .catch(next);
+}
+
+export function getProperty(req: Request, res: Response, next: NextFunction) {
+  return findProperty(req.params.propertyId)
+    .then(property => {
```

Comment:
Shouldn't this check that the property isn't undefined/null and throw a 404 if it is?

Reply:
Pretty sure the request won't get matched to this route I'd there's not properyId

Reply:
The route just accepts any id though. So you could pass anything in and we would return null.

Reply:
It's mostly just a pattern argument.